apps/dashboard/src/utils/liquid-autocomplete.ts

```diff
@@ -251,6 +257,17 @@ export function createAutocompleteSource(variables: LiquidVariable[], isEnhanced
           // Add 2 if we need to account for closing brackets
           const finalCursorPos = from + wrappedValue.length + (needsClosing ? 0 : 2);
 
+          if (completion.type === 'digest') {
```

Comment:
This code couples autocompletion with the tracking functionality for a specific control-input use case. Since this code is a critical piece of functionality without any testing yet, I suggest passing a callback `onSelect` function and then provide in the createAutocomplete function.